src/client/dfs/dfs.c

```diff
@@ -1075,7 +1074,9 @@ open_symlink(dfs_t *dfs, dfs_obj_t *parent, int flags, daos_oclass_id_t cid,
 		entry->value_len = value_len;
 		rc = insert_entry(parent->oh, DAOS_TX_NONE, sym->name, len,
 				  DAOS_COND_DKEY_INSERT, entry);
-		if (rc) {
+		if (rc == EEXIST) {
+			D_FREE(sym->value);
+		} else if (rc != 0) {
 			D_FREE(sym->value);
 			D_ERROR("Inserting entry %s failed (rc = %d)\n",
 				sym->name, rc);
```

Comment:
I think this is probably one of the fixes.
```suggestion
			D_ERROR("Inserting entry '%s' failed: %d (%s)\n", sym->name, rc, strerror(rc));
```

Reply:
Thank you so much for checking out fault injection issue and suggested the fix! Just committed your changes.